sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -920,6 +921,10 @@ TINYINT_LITERAL
     : DIGIT+ 'Y'
     ;
 
+BYTELENGTH_LITERAL
+    : DIGIT+ ('b' | 'B' | 'k' | 'K' | 'm' | 'M' | 'g' | 'G')
```

Comment:
The lexer is case insensitive so we don't need to add both lower and upper case characters. Just keep the uppercase ones (for consistency).